app/routes/post.js

```diff
@@ -1,35 +1,60 @@
+require('../../config/mongoose')
 const express = require('express')
 const router = express.Router()
 const userController = require('../controllers/post')
+const uploader = require('../utils/uploader')
+const auth = require('../middleware/auth')
 
 // CREATE A POST
 router.post(
   '/',
+  auth,
+  uploader.upload.single('imgUrl'),
   userController.create
 )
 
-// GET ALL POSTS OF A USER
 router.get(
-  '/',
+  '/auth',
   userController.authenticate
 )
 
-// GET PARTICULAR POST OF A USER
+// GET ALL POSTS
 router.get(
-  '/:id',
-  userController.test
+  '/all_posts',
+  auth,
+  userController.getAllPost
 )
 
 // UPDATE A TASK
 router.patch(
   '/:id',
-  userController.test
+  auth,
+  uploader.upload.single('imgUrl'),
+  userController.updatePost
 )
 
 // DELETE A TASK
 router.delete(
   '/:id',
-  userController.test
+  auth,
+  userController.delete
+)
+
+// GET TASK BY ID
+router.get(
+  '/:id',
+  auth,
+  userController.getPostById
+)
+
+router.put(
+  '/upvote/:id/:userId',
```

Comment:
this should be just `/upvote/:id`

Reply:
Then, how do we store the user who is making the upvote? Should I send both userID and the commentID in the same parameter and then split them?